Tests/ApolloCacheDependentTests/FetchQueryTests.swift

```diff
@@ -277,11 +314,26 @@ class FetchQueryTests: XCTestCase {
 
       let expectation = self.expectation(description: "Fetching query")
 
-      client.fetch(query: query, cachePolicy: .returnCacheDataDontFetch) { (result, error) in
+      client.fetch(query: query, cachePolicy: .returnCacheDataDontFetch) { outerResult in
         defer { expectation.fulfill() }
-
-        XCTAssertNil(error)
-        XCTAssertNil(result)
+        switch outerResult {
+        case .success:
+          XCTFail("This should have returned an error!")
+        case .failure(let error):
+          if
+            let resultError = error as? GraphQLResultError,
+            let underlyingError = resultError.underlying as? JSONDecodingError {
+              switch underlyingError {
+              case .missingValue:
+                // Correct error!
+                break
+              default:
+                XCTFail("Unexpected JSON error: \(error)")
+              }
+          } else {
+            XCTFail("Unexpected error: \(error)")
+          }
+        }
```

Comment:
This test also represents a breaking change - I *think* that this has an outer `GraphQLResultError` since there's some data in the cache, but not all of it.